trace_decoder/src/compact/compact_prestate_processing.rs

```diff
@@ -325,12 +328,8 @@ impl ParserState {
     fn parse(mut self) -> CompactParsingResult<WitnessOutput> {
         let mut entry_buf = Vec::new();
 
-        // TODO: Consider moving this into the `Self`...
-        let mut storage_tries = HashMap::new();
-
         loop {
-            let num_rules_applied =
-                self.apply_rules_to_witness_entries(&mut storage_tries, &mut entry_buf)?;
+            let num_rules_applied = self.apply_rules_to_witness_entries(&mut entry_buf)?;
```

Comment:
Not too familiar with this code, but doesn't `apply_rules_to_witness_entries` already apply all of the rules? Do we really need to `loop` here and check the number of rules applied again?

Reply:
Yeah that's an interesting thought. The truth is I don't know for sure, but I think you're right for all of the test payloads used in the unit tests.

The [spec seems to suggest](https://github.com/ledgerwatch/erigon/blob/devel/docs/programmers_guide/witness_formal_spec.md#execution-process) that multiple runs are possible in some cases. From the implementation of `apply_rules_to_witness_entries`, the cursor moves from `left` --> `right`, where it can skip an element if there's currently no rule to apply. Whenever it calls `apply_rules_to_witness_entries` it always starts at the beginning, potentially being a way to go back and check nodes that it had previously skipped. But yeah. As to whether or not this can actually occur in practice for some edge cases, I don't know.

Reply:
Going to leave this in just because the spec suggests that this can happen in theory.